girder/api/rest.py

```diff
@@ -227,6 +227,17 @@ def retVal(user, token):
         return retVal(user, token)
 
 
+def setCurrentUser(user):
```

Comment:
@zachmullen 

Since you introduced this method, could you update [api/v1/user.py](https://github.com/girder/girder/blob/master/girder/api/v1/user.py) and [api/rest.py](https://github.com/girder/girder/blob/master/girder/api/rest.py) to use this method instead of the calls to 

    setattr(cherrypy.request, 'girderUser', user)

for consistency and for the benefit of hiding the use of `cherrypy` a bit?

Perhaps also hide cherrypy in [these lines](https://github.com/girder/girder/blob/master/girder/api/rest.py#L131-L132), while you're at it.